### PR TITLE
AzureCP PR - Update AutoRest Compute Lib.

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

File: lib/commands/arm/armsdk/compute.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//
```
Comment (amarzavery) [inline lib/commands/arm/armsdk/compute.js]: Why do we need this file, if we have moved to autorest?

Comment (huangpf) [inline lib/commands/arm/armsdk/compute.js]: Good catch -- VMSS still needs it for a while. I would remove it after that is done.

File: lib/commands/arm/vm/vmImage._js [hunk 2]
```diff
@@ -122,30 +110,30 @@ __.extend(VMImage.prototype, {
...
-          image.dataDiskImages = imgResult.dataDiskImages;
-          image.purchasePlan = imgResult.purchasePlan;
-          return callBack(null, image);
-        });
+        that.serviceClients.computeManagementClient.virtualMachineImages.get(
+          imageFilter.location,
+          imageFilter.offer,
+          imageFilter.publishername,
+          imageFilter.skus,
+          image.name,
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmImage._js]: Why is this code duplicated at two places?

File: lib/commands/arm/vm/vmImage._js [hunk 3]
```diff
@@ -174,32 +162,32 @@ __.extend(VMImage.prototype, {
...
-            image.offer = imageFilter.offer;
-            image.skus = skus.vmImages.skus;
-            image.urn = image.publisher + ':' +image.offer + ':' + image.skus + ':' + image.name;
-            image.operatingSystem = imgResult.oSDiskImage.operatingSystem;
-            image.dataDiskImages = imgResult.dataDiskImages;
-            image.purchasePlan = imgResult.purchasePlan;
-            return callBack(null, image);
-          });
+          that.serviceClients.computeManagementClient.virtualMachineImages.get(
+            imageFilter.location,
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmImage._js]: This is the other place

File: lib/commands/arm/vm/vmImage._js [hunk 4]
```diff
@@ -249,48 +237,40 @@ __.extend(VMImage.prototype, {
...
-              image.publisher = imageFilter.publishername;
-              image.offer = skusCollection.offer;
-              image.skus = skus.name;
-              image.urn = image.publisher + ':' +image.offer + ':' + image.skus + ':' + image.name;
-              image.operatingSystem = imgResult.oSDiskImage.operatingSystem;
-              image.dataDiskImages = imgResult.dataDiskImages;
-              image.purchasePlan = imgResult.purchasePlan;
-              return callBack(null, image);
-            });
+            that.serviceClients.computeManagementClient.virtualMachineImages.get(
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmImage._js]: third place

File: lib/commands/arm/vm/vmImage._js [hunk 5]
```diff
@@ -347,19 +327,19 @@ __.extend(VMImage.prototype, {
...
-            type: typeItem.name
-          }, function(error, verItems) {
-            verItems = error ? { resources: [] } : verItems;
-            verItems.publisher = typeCollection.publisher;
-            verItems.location = typeCollection.location;
-            verItems.typeName = typeItem.name;
-            return callBack(error, verItems);
-          });
+          that.serviceClients.computeManagementClient.virtualMachineExtensionImages.listVersions(
+            typeCollection.location,
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmImage._js]: listVersions duplicated at two places.

File: lib/commands/arm/vm/vmImage._js [hunk 6]
```diff
@@ -380,21 +360,21 @@ __.extend(VMImage.prototype, {
...
     var verQueries = [];
-    typeNames.resources.forEach(function(typeItem) {
+    typeNames.forEach(function(typeItem) {
       verQueries.push(function(callBack) {
-        that.serviceClients.computeManagementClient.virtualMachineExtensionImages.listVersions({
-          location: location,
-          publisherName: publisher,
-          type: typeItem.name},
+        that.serviceClients.computeManagementClient.virtualMachineExtensionImages.listVersions(
+          location,
```
Comment (amarzavery) [inline lib/commands/arm/vm/vmImage._js]: other place for listVersions

**Pull request merged**